### PR TITLE
Mod2c as submodule

### PR DESCRIPTION
**Pull request by fouriaux**

### CODE REVIEW

**Comment by fouriaux** [comment]
I haven't changed the findmod2c as it is not possible to add ExternalProject_add in it without significant changes, that seems to be something aside to think about. Not on current path.

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[submodule "external/mod2c"]
+	path = external/mod2c
+  url = https://github.com/BlueBrain/mod2c
+  branch = coreneuron_submodule
```
Comment (pramodskumbhar) [inline .gitmodules]: to master

File: coreneuron/CMakeLists.txt [hunk 2]
```diff
@@ -71,7 +62,7 @@ macro(mod2c_target name input)
     list(APPEND MOD2C_${name}_MODS "${mod2c_modname_}")
 
     add_custom_command(OUTPUT "${mod2c_output_}"
-      DEPENDS ${input} "${mod2c_install_prefix}/mod2c_core"
+      DEPENDS ${input} mod2c
```
Comment (pramodskumbhar) [inline coreneuron/CMakeLists.txt]: could you check if change in mod2c recompile mod files?

Comment (fouriaux) [reply coreneuron/CMakeLists.txt]: It dont recompile when there is changes to mod2c

**Pull request merged**